lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -409,6 +409,78 @@ def test_empty(self):
             }
         )
 
+    def test_nonexistent_topic_id(self):
+        source_threads = [make_minimal_cs_thread()]
+        self.assertEqual(
+            self.get_thread_list(source_threads, topic_id="nonexistent_topic_id"),
+            {
+                "results": [],
+                "next": None,
+                "previous": None,
+            }
+        )
+
+    def test_get_threads_by_topic_id(self):
```

Comment:
I don't think we need to check thread content in this test. The really important thing here is that we are checking that the right URL is accessed. I think a sufficient test would be something like the following:

```
self.get_thread_list([], topic_id="test_topic")
self.assertEqual(urlparse(httpretty.last_request().path).path, "/api/v1/test_topic/threads")
```